allennlp/models/encoder_decoders/nlvr_semantic_parser.py

```diff
@@ -820,25 +1011,31 @@ def _compute_new_states(cls,  # type: ignore
                     continue
                 # If action is not in instance_agenda, mask_variable, and checklist_addition will be
                 # all 0s.
-                checklist_mask = instance_agenda == action  # (agenda_size,)
-                checklist_addition = checklist_mask.float() * action_prob  # (agenda_size,)
+                checklist_mask = (instance_agenda == action).float()  # (agenda_size,)
+                agenda_action_prob = torch.sum(checklist_mask * agenda_action_probs)  # (1,)
+                # TODO (pradeep): This is not a great way to bias the model towards choosing actions
```

Comment:
One other option: see equation 11 here: https://arxiv.org/pdf/1704.04368.pdf.

Reply:
Great! Thanks for the pointer. This is exactly what I meant by "use the current checklist in the action logit computation" in the comment there.